base/sparse/sparsematrix.jl

```diff
@@ -134,30 +134,54 @@ function Base.show(io::IOContext, S::SparseMatrixCSC)
     end
 
     limit::Bool = get(io, :limit, false)
-    if limit
-        rows = displaysize(io)[1]
-        half_screen_rows = div(rows - 8, 2)
+    rows = displaysize(io)[1] - 4 # -4 from [Prompt, header, newline after elements, new prompt]
+    rows <= 2 && return
```

Comment:
Ah that looks like a bug, something that should be fixed, not continued. It doesn't make any sense to let this ever be smaller than 2. Maybe print only the vdots if the limit is smaller than showing 3 lines